specs/capri/Multi-attach.md

```diff
@@ -0,0 +1,187 @@
+3. Add multiAttach bool to the volume model to enables/disables the ability of multi-attach. 
+4. Add 'multiAttach': '<is> True' to the profile and do some corresponding changes to the controller to choose driver if user requires a multi-attach volume.
+
+### Data model impact
+
+1. Add MultiAttach field in pool spec
+
+```
+type StoragePoolSpec struct {
+	*BaseModel
+	Name string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	Status string `json:"status,omitempty"`
+	DockId string `json:"dockId,omitempty"`
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+	TotalCapacity int64 `json:"totalCapacity,omitempty"`
+	FreeCapacity int64 `json:"freeCapacity,omitempty"`
+	StorageType string `json:"storageType,omitempty"`
+	Extras StoragePoolExtraSpec `json:"extras,omitempty"`
+	ReplicationType string `json:"replicationType,omitempty"`
+	ReplicationDriverName string `json:"replicationDriverName,omitempty"`
+
+	// MultiAttach
+	// If true, this volume can attach to more than one instance. Default will be multiAttach:False
+	MultiAttach bool `json:"multiAttach,omitempty"`
+}
+```
+
+
+2. The volume attachment struct adds AttachMode, read-only (‘ro’) or read-and-write (‘rw’), default is ‘rw’.
+
+```
+type VolumeAttachmentSpec struct {
+	*BaseModel
+	TenantId string `json:"tenantId,omitempty"`
+	UserId string `json:"userId,omitempty"`
+	VolumeId string `json:"volumeId,omitempty"`
+	Mountpoint string `json:"mountpoint,omitempty"`
+	Status string `json:"status,omitempty"`
+	Metadata map[string]string `json:"metadata,omitempty"`
+	HostInfo `json:"hostInfo,omitempty"`
+	ConnectionInfo `json:"connectionInfo,omitempty"`
+	AccessProtocol string `json:"accessProtocol,omitempty"`
+	AttachMode string `json:"attachMode,omitempty"`
+}
+```
+
+3. The volume struct adds MultiAttach.
+
+```
+type VolumeSpec struct {
+	*BaseModel
+	TenantId string `json:"tenantId,omitempty"`
+	UserId string `json:"userId,omitempty"`
+	Name string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	GroupId string `json:"groupId,omitempty"`
+	Size int64 `json:"size,omitempty"`
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+	Status string `json:"status,omitempty"`
+	PoolId string `json:"poolId,omitempty"`
+	ProfileId string `json:"profileId,omitempty"`
+	Metadata map[string]string `json:"metadata,omitempty"`
+	SnapshotId string `json:"snapshotId,omitempty"`
+	SnapshotFromCloud bool `json:"snapshotFromCloud,omitempty"`
+	ReplicationId string `json:"replicationId,omitempty"`
+	ReplicationDriverData map[string]string `json:"replicationDriverData,omitempty"`
+	AttachStatus string
+	MultiAttach bool `json:"multiAttach"`
+}
+```
+
+### REST API impact
+
+N/A
+
+### Security impact
+
+N/A
+
+### Other end user impact
+
+N/A
```

Comment:
User needs to configure the `multiAttach` flag for their backend pools, which should be end user impact.

Reply:
Done.